presto-main/src/main/java/com/facebook/presto/sql/planner/BaseExtractor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.sql.tree.DefaultExpressionTraversalVisitor;
+import com.facebook.presto.sql.tree.DereferenceExpression;
+import com.facebook.presto.sql.tree.Identifier;
+import com.facebook.presto.sql.tree.SubscriptExpression;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+public class BaseExtractor
+        extends DefaultExpressionTraversalVisitor<Symbol, Void>
+{
+    @Override
+    protected Symbol visitIdentifier(Identifier node, Void context)
+    {
+        return new Symbol(node.getValue());
```

Comment:
It is a bad idea. Identifier belongs to AST and symbol to IR (plan node tree) model.

Reply:
the class is removed.